File: app/src/main/java/com/example/msimple/data/model/ICategoryModel.java
package com.example.msimple.data.model;

import com.example.msimple.data.vos.CategoryVO;
import com.example.msimple.data.vos.ProgramVO;

import java.util.List;

public interface ICategoryModel {
    void getCategory(Response response);
    ProgramVO getProgramVO(String id);

    interface Response{

        void onSuccess(List<CategoryVO> categoryVO);
        void onFail(String message);

    }
}

File: app/src/main/java/com/example/msimple/data/model/ICategoryModel.kt
package com.example.msimple.data.model

import com.example.msimple.data.vos.CategoryVO
import com.example.msimple.data.vos.ProgramVO

interface ICategoryModel {
    fun getCategory(response: Response)
    fun getProgramVO(id: String): ProgramVO

    interface Response {

        fun onSuccess(categoryVO: List<CategoryVO>)
        fun onFail(message: String)

    }
}
